src/card_order/index.ts

```diff
@@ -0,0 +1,121 @@
+import { ApiPostError, ApiGetError } from "../errors";
+
+import { getApiServer } from "../meta";
+import { Oauth } from "../user"
+
+import { BasicUser } from "../user";
+
+export enum OrderStates {
+    CREATED     = "CREATED",
+    IN_PROGRESS = "IN_PROGRESS",
+    FINISHED    = "FINISHED",
+    CANCELLED   = "CANCELLED",
+}
+
+interface BaseCardOrder {
+    uuid: string;
+    
+    created: number;
+    state: OrderStates
+    last_updated: number;
+}
+
+export type BasicCardOrder = BaseCardOrder & {
+    event_uuid: string;
+    subject_user: BasicUser;
+    creator_user: BasicUser;
+    
+    updated_by_user: BasicUser;
+}
+
+//? init
+export const createCardOrder = async (user_uuid: string) => {
+    const response = await fetch(`${getApiServer()}/card_order`, {
+        method: "POST",
+        headers: {
+			"Content-Type": "application/json",
+			...(await Oauth.getAuthHeaders())
+		},
+        body: JSON.stringify({
+            user_uuid: user_uuid
+        })
+    });
+
+    if (!response.ok) {
+            let error = ""
+            try {
+                error = (await response.json())["error"]
+            } catch (e) {
+                throw new ApiPostError("Unable to create card order");
+            }
+
+            throw new ApiPostError(error);
+        }
+    return await response.json() as BasicCardOrder;
+}
+
+export const getAllCardOrders = async (event_uuid:string|null) => {
+    const response = await fetch(`${getApiServer()}/card_order/?event_uuid=${event_uuid}`, { // Idk if this works
```

Comment:
This might possibly be a bad request if event_uuid is null since we still pass the empty parameter